plugins/osdn/ovs/controller.go

```
@@ -21,26 +33,224 @@ func NewFlowController(multitenant bool) *FlowController {
+		return false
+	}
+
+	return true
+}
+
+func deleteLocalSubnetRoute(device, localSubnetCIDR string) error {
+	const (
+		timeInterval = 100 * time.Millisecond
+		maxIntervals = 20
+	)
+
+	for i := 0; i < maxIntervals; i++ {
+		itx := ipcmd.NewTransaction(string)
+		routes, err := itx.GetRoutes()
+		if err != nil {
+			return err
+		}
+		for _, route := range routes {
+			if route.Contains(localSubnetCIDR) && route.Contains("dev "+device) {
+				itx.DeleteRoute(localSubnetCIDR + " dev " + device)
+				return itx.EndTransaction()
+			}
+		}
+
+		time.Sleep(timeInterval)
+	}
+
+	return fmt.Errorf("Could not find %s route for dev %s", localSubnetCIDR, device)
+}
+
 func (c *FlowController) Setup(localSubnetCIDR, clusterNetworkCIDR, servicesNetworkCIDR string, mtu uint) error {
 	_, ipnet, err := net.ParseCIDR(localSubnetCIDR)
 	localSubnetMaskLength, _ := ipnet.Mask.Size()
 	localSubnetGateway := netutils.GenerateDefaultGateway(ipnet).String()
-	out, err := exec.Command("openshift-sdn-ovs-setup.sh", localSubnetGateway, localSubnetCIDR, fmt.Sprint(localSubnetMaskLength), clusterNetworkCIDR, servicesNetworkCIDR, fmt.Sprint(mtu), fmt.Sprint(c.multitenant)).CombinedOutput()
-	if err != nil {
-		glog.Infof("Output of setup script:\n%s", out)
-		exitErr, ok := err.(*exec.ExitError)
-		if ok {
-			status := exitErr.ProcessState.Sys().(syscall.WaitStatus)
-			if status.Exited() && status.ExitStatus() == 140 {
-				// valid, do nothing, its just a benevolent restart
-				return nil
-			}
-		}
-		glog.Errorf("Error executing setup script: %v\n", err)
+
+	itx = ipcmd.NewTransaction(LBR)
+	itx.SetLink("down")
+	itx.IgnoreError()
+	itx.DeleteLink()
+	itx.IgnoreError()
+	itx.AddLink("type", "bridge")
+	itx.AddAddress(fmt.Sprintf("%s/%d", localSubnetGateway, localSubnetMaskLength))
+	itx.SetLink("up")
+	err = itx.EndTransaction()
+	if err != nil {
+		log.Errorf("Failed to configure docker bridge: %v", err)
+		return err
+	}
+	defer deleteLocalSubnetRoute(LBR, localSubnetCIDR)
```

We ignore the error from deleteLocalSubnetRoute()
It might be useful to log the error in deleteLocalSubnetRoute() instead of returning the error?